### DISCUSSION TITLE
Allow zooming out beyond the length of the current clip

### DISCUSSION THREAD

**Discussion by trappar** [SynthstromAudible/DelugeFirmware#314] [Ideas]
This limitation has always seemed strange to me. I'd like to zoom out so I can more quickly manipulate the length of clips. Like if I know that I want to make an 8-bar four chord progression, then what I'd like to do is:
1. Zoom out to 2nd-notes
2. Increase the clip length to 8 bars
3. Start entering my chords

I know I could just double the length of the clip three times, but then if I've already started entering notes I get junk I don't want. I just don't see any compelling reason why this limitation *should* exist. There's already existing UI to show that part of a visible area is not in the clip.

Also, I think this would help when shortening clips since currently it doing so will constantly changes the time division. I find this jarring and I have to reorient myself after doing it. Seems way easier mentally if the zoom level only changes when you actually change it.

**Comment by weavermedia** [comment]
I don't use the length-doubling feature for extending clips for that same reason - I often don't want the notes to be duplicated.

I use SHIFT+TURN<> to lengthen the clip to 8 bars. That just adds extra length without duplicating notes. Also, it keeps me at the same zoom level I was originally work at, which I prefer.

Your suggestion to zoom out, extend and enter new chords probably needs a 2b step of zooming back in before you start entering chords. Unless you're entering them at 2nd note resolution I suppose.

**Reply by trappar** [reply]
Yeah I'm saying I like being able to input chords at 2nd note resolution.

And yes, I can extend the clip length first, it just means I have to scroll the <> knob much further since it scrolls slower when zoomed in than when zoomed out.

Again, why have the limitation? I know there are plenty of ways to work around it but I can't think of any good reason why it even exists.

**Reply by weavermedia** [reply]
Cools. I wasn't saying it was a bad idea, just wanted to make sure you knew the options.

The action you're describing doesn't seem to be much more efficient than extending length with SHFT then zooming out to 2nd notes.

**Comment by weavermedia** [comment]
If you zoomed out a 1-bar clip to 2nd notes would you expect to be able to place notes on the pads beyond the current 1-bar length of the clip? If so, would that automatically extend the clip length to that point?

**Reply by trappar** [reply]
I was thinking you'd just be able to input notes in the existing clip bounds, but it does seem like it would be a really nice quality of life feature to allow it to extend however far it needed to for new out-of-bound notes.

**Reply by weavermedia** [reply]
When you zoom out past the clip's original length, would the bounds of the clip be visible in any way? Meaning, how would you know where the end of the clip is when you're zoom out?

**Reply by trappar** [reply]
It would use the existing way of showing that. Blur the out of bound section. Like this:

![image](https://github.com/SynthstromAudible/DelugeFirmware/assets/525726/26bcf196-0aea-4e98-a41d-ff48af70165c)

In that image only the first 4 steps are editable.

**Reply by weavermedia** [reply]
Would you tap a blurred pad to add a note there and automatically extend the clip length?
